amplitude.js

```diff
@@ -5864,6 +5865,29 @@ AmplitudeClient.prototype._apiKeySet = function _apiKeySet(methodName) {
  */
 AmplitudeClient.prototype._loadSavedUnsentEvents = function _loadSavedUnsentEvents(unsentKey) {
   var savedUnsentEventsString = this._getFromStorage(localStorage$1, unsentKey);
+  var events = this._parseSavedUnsentEventsString(savedUnsentEventsString, unsentKey);
+
+  var savedUnsentEventsStringLegacy = this._getFromStorageLegacy(localStorage$1, unsentKey);
+  var legacyEvents = this._parseSavedUnsentEventsString(savedUnsentEventsStringLegacy, unsentKey);
+
+  var unsentEvents = events.concat(legacyEvents);
```

Comment:
You should put the legacy ones in front of the new ones so that the older ones come first.

Reply:
If you really wanted to you could also sort by the sequence number or event id for a true ordering (but probably not necessary - note sequence number was only added in later versions of the SDK: https://github.com/amplitude/Amplitude-JavaScript/blob/master/src/amplitude-client.js#L1164-L1166)